fitbenchmarking/utils/fitbm_result.py

```
@@ -8,22 +10,101 @@ class FittingResult(object):
+                output = "{:.4g}".format(self.value)
+            elif self.options.comparison_mode == "both":
+                output = "{:.4g} ({:.4g})".format(self.value, self.norm_value)
+        else:
+            output = 'Fitting problem class: minimizer = {0}'.format(
+                self.minimizer)
+        return output
+
+    @property
+    def table_type(self):
+        return self._table_type
+
+    @table_type.setter
+    def table_type(self, value):
+        self._table_type = value
+        if value == "runtime":
+            self.value = self.runtime
+            self.norm_value = self.norm_runtime
+            self.colour = self.colour_runtime
+        if value == "acc":
+            self.value = self.chi_sq
+            self.norm_value = self.norm_acc
+            self.colour = self.colour_acc
+
+    def set_colour_scale(self):
+        """
+        Utility function set colour rendering for html tables
+        """
+        colour_scale = self.options.colour_scale
+        colour_bounds = [colour[0] for colour in colour_scale]
+        # prepending 0 value for colour bound
+        colour_bounds = [0] + colour_bounds
+        html_colours = [colour[1] for colour in colour_scale]
+        self.colour_runtime = colour_scale[-1]
+        self.colour_acc = colour_scale[-1]
+        for i in range(len(colour_bounds) - 1):
+            if colour_bounds[i] < self.norm_runtime <= colour_bounds[i + 1]:
+                self.colour_runtime = html_colours[i]
+            if colour_bounds[i] < self.norm_acc <= colour_bounds[i + 1]:
+                self.colour_acc = html_colours[i]
+
+    @property
+    def min_chi_sq(self):
+        return self._min_chi_sq
+
+    @min_chi_sq.setter
+    def min_chi_sq(self, value):
+        self._min_chi_sq = value
+        try:
+            if not self.chi_sq > 0:
+                self.chi_sq = np.inf
+            self.norm_acc = self.chi_sq / self.min_chi_sq
+        except Exception as excp:
```

If this is divide by zero we should catch only those exceptions.
We should also either raise an exception or return a sensible value.

---

it's a  `ZeroDivisionError`

---

I don't think the try except is needed so it has been removed